README.md

```diff
@@ -1,26 +1,270 @@
+- [Dependabot](https://github.com/dependabot) commits that happened after the last release.
+- [Renovate](https://github.com/renovatebot/renovate) commits that happened after the last release.
+
+The changelog object represented in this YAML file has 3 important fields:
+- `changes`: List of changes parsed from `CHANGELOG.md` Changes have a type and a message, and changes belonging to the same type are grouped in the release notes.
+- `dependencies`: List of dependencies that have been updated in this release. Entries optionally include from and to which version the dependency has been updates.
+- `notes`: This is a free-text field that is included in the release notes before the list of changes. It can be used to add context to some changes, point users to documentations, or anything at all. Markdown syntax is allowed.
+
+A freshly-generated `changelog.yaml` file looks like this:
+
+```yaml
+notes: |-
+  ### Important announcement (note)
+  This is a release note
+changes:
+  - type: breaking
+    message: Support has been removed
+  - type: security
+    message: Fixed a security issue
+  - type: enhancement
+    message: New feature has been added
+dependencies:
+  - name: docker/setup-buildx-action
+    from: "1.0.0"
+    to: "2.0.0"
+    meta:
+      pr: "100"
+      commit: a72b98709dfa0d28cf7c73020f3dede670f7a37f
+```
+
+In addition to the fields listed above, there's a fourth boolean field called `is-held`. Purpose and behavior of this field is explained in the advanced section of this manual.
+
+This `generate-yaml` command will extract this changes and notes from 3 different sources:
+
+## `render-markdown` and `update-markdown`
+
+<img align="right" width="40%" alt="generate-yaml action diagram" src="https://user-images.githubusercontent.com/969721/195116114-a35ba8c7-1242-4c99-9c49-b7c627e7a9c2.png">
+
+As a part of the release process, contents of the `## Unreleased` section in `CHANGELOG.md` need to be moved under a header with the name of the version being released. This is done by `render-markdown`, which loads a `changelog.yaml` file and pretty-prints it in Markdown format, grouping changes under headers depending on their type.
+
+The list of entry types, and therefore groups, supported by the release-toolkit are:
+
+- `breaking`: A breaking change on a user-facing API. Rendered first and with a warning sign to quickly catch the attention of the reader.
+- `security`: Security fixes that remediate potential or existing security vulnerabilities.
+- `enhancement`: New features or improvements to existing ones.
+- `bugfix`: Fixes to incorrect behavior of the application.
+
+Additionally, a section with changes to dependencies is also included after the list of changes.
+
+The Markdown version of the `changelog.yaml` file mentioned above would be:
+
+```markdown
+## v1.2.3 - 2022-10-11
+
+### Important announcement (note)
```

Comment:
how this important note is added? Is there an entry type for this ?

Reply:
This will be added as an additional note exactly as it is